Source/Charts/Data/Implementations/Standard/ChartDataEntry.swift

```diff
@@ -108,32 +99,18 @@ open class ChartDataEntry: ChartDataEntryBase
     }
 }
 
-public func ==(lhs: ChartDataEntry, rhs: ChartDataEntry) -> Bool
-{
-    if lhs === rhs
-    {
-        return true
-    }
-    
-    if !lhs.isKind(of: type(of: rhs))
+// MARK: Equatable
+extension ChartDataEntry/*: Equatable*/ {
+    public static func ==(lhs: ChartDataEntry, rhs: ChartDataEntry) -> Bool
     {
-        return false
-    }
-    
-    if lhs.data !== rhs.data && !lhs.data!.isEqual(rhs.data)
-    {
-        return false
-    }
-    
-    if fabs(lhs.x - rhs.x) > Double.ulpOfOne
-    {
-        return false
-    }
-    
-    if fabs(lhs.y - rhs.y) > Double.ulpOfOne
-    {
-        return false
+        if lhs === rhs
+        {
+            return true
+        }
+
+        return lhs.isKind(of: type(of: rhs))
+            && (lhs.data?.isEqual(rhs.data) ?? true)
```

Comment:
The final section of this [API Doc](https://developer.apple.com/documentation/swift/equatable) explains the difference between equality and identity, and why identity should not be considered when implementing `Equatable`. There is an argument to be made for the first check (`if lhs === rhs { return true }`), because it allows a quick escape without having to check each parameter individually (though I still disagree with it. I'm curious to see the performance of that identity check vs simply checking the equality of the ivars).

With data there are 5 possible scenarios
```
   lhs, rhs
1. nil, nil
2. val1, nil
3. nil, val1
4. val1, val2
5. val1, val1 
```
using `lhs.data?.isEqual(rhs.data) ?? false` will provide the correct result in cases 2, 3, 4, and 5. In order to account for case 1, we must also check to see if they are both `nil`.